packages/pages/src/dev/server/middleware/indexPage.ts

```diff
@@ -107,6 +113,33 @@ export const indexPage =
         );
       }
 
+      if (serverlessFuncs.length) {
```

Comment:
Could we move this new logic into a separate function? I was thinking we could even move all the serverless funcs stuff into a separate file but I'm also fine keeping it here.
Imo each of the preexisting indexPageHtml.replace calls should have been separated out instead of this giant templating script (out of scope though!)
Eventually I think we'll want to refactor this index page to use react, it feels a little backward to have a js library that renders react pages but uses raw js templating to create the index page.

Reply:
I made an item to update the index page to use react https://yexttest.atlassian.net/browse/SLAP-2642

Reply:
Moved it into a separate function, but kept it in the same file for now so we can refactor it all from one place